### PR TITLE
callbacks replaced with await, deleted unused imports

### PR DESCRIPTION
**Pull request by kmkcc2**

### CODE REVIEW

File: controllers/user.controller.js [hunk 1]
```diff
@@ -22,101 +21,93 @@ exports.create = (req, res) => {
...
+        message: "User was deleted successfully!",
+      });
+    } else {
+      return res.send({
+        message: `Cannot delete User with id=${id}. Maybe User was not found!`,
       });
+    }
+  } catch (err) {
+    return res.status(500).send({
+      message: "Could not delete User with id=" + id,
```
Comment (FilipAlbrycht) [inline controllers/user.controller.js]: IMHO this is not proper message for internal server error. I think you should keep 500 as HTTP code but message should be changed to more generic like "Internal Server Error" and error itself should be displayed in logs

File: controllers/user.controller.js [hunk 2]
```diff
@@ -22,101 +21,93 @@ exports.create = (req, res) => {
...
-    .catch((err) => {
-      res.status(500).send({
-        message: "Could not delete User with id=" + id,
+exports.destroy = async (req, res) => {
+  try {
+    const id = req.params.id;
+    const response = await User.destroy({
+      where: { id: id },
+    });
+    if (response == 1) {
```
Comment (FilipAlbrycht) [inline controllers/user.controller.js]: === instead of == - read about difference

File: controllers/user.controller.js [hunk 3]
```diff
@@ -22,101 +21,93 @@ exports.create = (req, res) => {
...
+      return res.send({
+        message: "User was deleted successfully!",
+      });
+    } else {
+      return res.send({
```
Comment (FilipAlbrycht) [inline controllers/user.controller.js]: so it means that user was not found so you should throw 404 HTTP code

File: controllers/user.controller.js [hunk 4]
```diff
@@ -22,101 +21,93 @@ exports.create = (req, res) => {
...
-          message: `Cannot delete User with id=${id}. Maybe User was not found!`,
-        });
-      }
-    })
-    .catch((err) => {
-      res.status(500).send({
-        message: "Could not delete User with id=" + id,
+exports.destroy = async (req, res) => {
+  try {
+    const id = req.params.id;
```
Comment (FilipAlbrycht) [inline controllers/user.controller.js:127]: check if id acutally exists in params

File: controllers/user.controller.js [hunk 5]
```diff
@@ -22,101 +21,93 @@ exports.create = (req, res) => {
...
-        });
-      }
-    })
-    .catch((err) => {
-      res.status(500).send({
-        message: "Error updating User with id=" + id,
+exports.update = async (req, res) => {
+  try {
+    const id = req.params.id;
+    const response = await User.update(req.body, {
```
Comment (FilipAlbrycht) [inline controllers/user.controller.js]: You cannot pass whole body to update. Extract object to make sure that you're passing only needed properties

File: controllers/user.controller.js [hunk 6]
```diff
@@ -22,101 +21,93 @@ exports.create = (req, res) => {
...
+      where: { id: id },
```
Comment (FilipAlbrycht) [inline controllers/user.controller.js]: shorter: where: { id }

File: controllers/user.controller.js [hunk 7]
```diff
@@ -22,101 +21,93 @@ exports.create = (req, res) => {
...
+    });
+    if (response == 1) {
```
Comment (FilipAlbrycht) [inline controllers/user.controller.js]: === instead of ==

Comment (FilipAlbrycht) [inline controllers/user.controller.js]: also this looks weird, is there any better way to check result?

File: controllers/user.controller.js [hunk 8]
```diff
@@ -22,101 +21,93 @@ exports.create = (req, res) => {
...
+      return res.send({
+        message: "User was updated successfully.",
+      });
+    } else {
+      return res.send({
+        message: `Cannot update User with id=${id}. Maybe User was not found or req.body is empty!`,
```
Comment (FilipAlbrycht) [inline controllers/user.controller.js]: looks like error so send proper http code :)

File: controllers/user.controller.js [hunk 9]
```diff
@@ -22,101 +21,93 @@ exports.create = (req, res) => {
...
+    const user = await User.findByPk(id);
+    if (user) return res.send(user);
+    else {
+      return res.status(404).send({
+        message: `Cannot find user with id: ${id}`,
       });
+    }
+  } catch (err) {
+    return res.status(500).send({
+      message: err.message || "Some error occurred while looking for user.",
```
Comment (FilipAlbrycht) [inline controllers/user.controller.js]: Internal Server Error

File: controllers/user.controller.js [hunk 10]
```diff
@@ -22,101 +21,93 @@ exports.create = (req, res) => {
...
-      res.send(data);
-    })
-    .catch((err) => {
-      res.status(500).send({
-        message:
-          err.message || "Some error occurred while while looking for users.",
-      });
+exports.findAll = async (req, res) => {
+  try {
+    const users = await User.findAll();
```
Comment (FilipAlbrycht) [inline controllers/user.controller.js]: res.send(await User.findAll())

File: controllers/user.controller.js [hunk 11]
```diff
@@ -22,101 +21,93 @@ exports.create = (req, res) => {
...
-          res.send(data);
-        })
-        .catch((err) => {
-          res.status(500).send({
-            message: err.message || "Some error occurred while creating new user.",
-          });
-        });
+  try {
+    const userUniqueEmailCheck = await User.findAll({
+      where: { email: user.email },
```
Comment (FilipAlbrycht) [inline controllers/user.controller.js:27]: if you already have unique contraint, you dont need to check that... but you can if you want :)

**Review by FilipAlbrycht — CHANGES_REQUESTED** [review]

File: controllers/user.controller.js [hunk 12]
```diff
@@ -22,101 +21,93 @@ exports.create = (req, res) => {
...
+    });
+    if (userUniqueEmailCheck.length > 0) {
+      return res.status(422).send({
+        message: "Email has already been taken.",
+      });
+    } else {
```
Comment (benmotyka) [inline controllers/user.controller.js]: jak cos jest zwracane wyzej to nie ma potrzeby tutaj pisania else, bo return konczy wywolanie funkcji

File: controllers/user.controller.js (hunk 10 above)
Comment (benmotyka) [reply controllers/user.controller.js]: imo to juz kwestia dyskusyjna, tak jak zrobil Kacper jest moim zdaniem czytelniej

File: controllers/user.controller.js [hunk 13]
```diff
@@ -22,101 +21,93 @@ exports.create = (req, res) => {
...
-          message: `Cannot find user with id: ${id}`,
-        });
-      }
-    })
-    .catch((err) => {
-      res.status(500).send({
-        message: err.message || "Some error occurred while looking for user.",
+exports.findOne = async (req, res) => {
+  try {
+    const id = req.params.id;
```
Comment (benmotyka) [inline controllers/user.controller.js:57]: dobra praktyka jest na poczatku funkcji pobierac parametry z zapytania tak jak to zrobiles

w sytuacji gdy potrzebowalbys wyciagnac wiecej parametrow, albo jakies propertisy z body, wtedy warto wykorzystac mechanizm destrukturyzacji przedstawiony w ES6 (https://kursjs.pl/kurs/obiekty/destructuring)

File: controllers/user.controller.js [hunk 14]
```diff
@@ -22,101 +21,93 @@ exports.create = (req, res) => {
...
+    const user = await User.findByPk(id);
+    if (user) return res.send(user);
+    else {
```
Comment (benmotyka) [inline controllers/user.controller.js]: jak wyzej cos zwrcasz to jest juz koniec funkcji i nie ma potrzeby pisania elsa

File: controllers/user.controller.js (hunk 6 above)
Comment (benmotyka) [reply controllers/user.controller.js]: https://javascript.plainenglish.io/object-literals-using-object-property-shorthand-6360825c60ef

File: controllers/user.controller.js (hunk 7 above)
Comment (benmotyka) [reply controllers/user.controller.js]: generalnie rzadko kiedy sie uzywa ==

File: controllers/user.controller.js [hunk 15]
```diff
@@ -22,101 +21,93 @@ exports.create = (req, res) => {
...
+  try {
+    const id = req.params.id;
+    const response = await User.update(req.body, {
+      where: { id: id },
+    });
+    if (response == 1) {
+      return res.send({
+        message: "User was updated successfully.",
+      });
+    } else {
```
Comment (benmotyka) [inline controllers/user.controller.js]: else zbedne

File: controllers/user.controller.js [hunk 16]
```diff
@@ -22,101 +21,93 @@ exports.create = (req, res) => {
...
-      }
-    })
-    .catch((err) => {
-      res.status(500).send({
-        message: "Could not delete User with id=" + id,
+exports.destroy = async (req, res) => {
+  try {
+    const id = req.params.id;
+    const response = await User.destroy({
+      where: { id: id },
```
Comment (benmotyka) [inline controllers/user.controller.js]: mozna skrocic zapis

File: controllers/user.controller.js (hunk 2 above)
Comment (kmkcc2) [reply controllers/user.controller.js]: respons in my case is an array with one value, thats why I chose the '==' sign. I can change it to `response[0] === 1`, but I will try to find a better way.

File: controllers/user.controller.js (hunk 11 above)
Comment (kmkcc2) [reply controllers/user.controller.js:27]: I did that, bcs I wanted to send more specific error message then `Validation Error`. I couldn't find a better way then checking it again.

File: controllers/user.controller.js (hunk 2 above)
Comment (kmkcc2) [reply controllers/user.controller.js]: seems like `if(response)` works just fine :sweat_smile:

File: controllers/user.controller.js [hunk 17]
```diff
@@ -22,101 +22,129 @@ exports.create = (req, res) => {
...
-        });
-      }
-    })
-    .catch((err) => {
-      res.status(500).send({
-        message: "Could not delete User with id=" + id,
+exports.destroy = async (req, res) => {
+  try {
+    const id = req.params.id;
+    if (isNaN(id) || id === " " || id === "") {
```
Comment (FilipAlbrycht) [inline controllers/user.controller.js:128]: not sure about this if statement, i think if (!id) will be enough

File: controllers/user.controller.js [hunk 18]
```diff
@@ -22,101 +22,129 @@ exports.create = (req, res) => {
...
-        });
-      }
-    })
-    .catch((err) => {
-      res.status(500).send({
-        message: err.message || "Some error occurred while looking for user.",
+exports.update = async (req, res) => {
+  try {
+    const id = req.params.id;
+    if (isNaN(id) || id === " " || id === "") {
```
Comment (FilipAlbrycht) [inline controllers/user.controller.js]: not sure about this if statement, i think if (!id) will be enough

**Review by FilipAlbrycht — APPROVED** [review]

**Review by benmotyka — APPROVED** [review]

**Pull request merged**